temporal/worker.go

```diff
@@ -0,0 +1,84 @@
+package temporal
+
+import (
+	"context"
+
+	"github.com/formancehq/go-libs/logging"
+	"go.temporal.io/sdk/activity"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/worker"
+	temporalworkflow "go.temporal.io/sdk/workflow"
+	"go.uber.org/fx"
+)
+
+type Definition struct {
+	Func any
+	Name string
+}
+
+type DefinitionSet []Definition
+
+func NewDefinitionSet() DefinitionSet {
+	return DefinitionSet{}
+}
+
+func (d DefinitionSet) Append(definition Definition) DefinitionSet {
+	d = append(d, definition)
+
+	return d
+}
+
+func New(logger logging.Logger, c client.Client, taskQueue string, workflows, activities []DefinitionSet, options worker.Options) worker.Worker {
+	options.BackgroundActivityContext = logging.ContextWithLogger(context.Background(), logger)
```

Comment:
Can you try to pass the context from the caller please?

Reply:
(fx provide one)

Reply:
Fixed thanks !